src/client/models/Notification.ts

```diff
@@ -0,0 +1,105 @@
+import {Entity} from './Entity';
+
+/**
+ * User-defined context data that can be attached to notifications
+ */
+export type NotificationContext = any;  // tslint:disable-line:no-any
+
+export enum NotificationTypes {
+    DEFAULT = 'DEFAULT',
+    BUTTON = 'BUTTON',
+    INLINE = 'INLINE',
+    INLINEBUTTON = 'INLINEBUTTON'
+}
+
+/**
+ * @description Interface for notification options
+ */
+export interface Notification extends Entity {
+    body: string;
+    title: string;
+    subtitle: string;
+    icon: string;
+    context: NotificationContext;
+    date: Date;
+    buttons: OptionButton[];
+    inputs: OptionInput[];
+    type?: NotificationTypes;
+}
+
+/**
+ * @description Options for notifications
+ */
+export interface NotificationOptions {
+    body: string;
+    title: string;
+    subtitle: string;
+    icon: string;
+    context: NotificationContext;
+    date: Date;
+    buttons: OptionButton[];
+    inputs: OptionInput[];
+}
+
+/**
+ * @description interface for notification event
+ */
+export interface NotificationEvent {
+    name: string;
+    uuid: string;
+    id: string;
+    context: NotificationContext;
+    buttons: OptionButton[];
+    buttonIndex: number;
+}
+
+/**
+ * @description Interface for buttons
+ */
+export interface OptionButton {
+    title: string;
+    icon: string;
+}
+
+/**
+ * @description Interface for inputs
+ */
+export interface OptionInput {
+    name: string;
+    placeholder: string;
+}
+
+/**
+ * @description This gets sent on every request the client makes to the service.
+ */
+export interface SenderInfo {
+    entityType: string;
+    name: string;
+    parentFrame: string;
+    uuid: string;
+    channelId: string;
+    channelName: string;
+}
+
+
+export interface INotification extends Notification {
```

Comment:
No worries if the answer is 'we'll worry about this later', but any way to change the naming to get rid of the `I` prefix here?

Reply:
`INotification` needs to go and probably to be joined with `SenderInfo` but thats outside of the scope of this PR.